text/069-coq-roadmap.md

```diff
@@ -0,0 +1,316 @@
+- Title: Coq roadmap
+
+- Drivers: Nicolas Tabareau (@tabareau), Théo Zimmermann (@Zimmi48)
+
+----
+
+# Summary
+
+This CEP aims to establish a roadmap for the Coq project.
+It highlights both:
+
+- what are considered as important axes to work on in the future,
+- what resources are available to work on these axes, and which axes we commit
+  to work on based on these resources.
+
+Resources mean availability of persons to conduct the work. For technical axes,
+requiring changes in the Coq system, it includes availability of persons to
+review and merge the changes.
+
+# Motivation
+
+Producing a roadmap for the Coq project is important for several reasons:
+
+- It helps the community to know what to expect from the Coq project in the
+  future, in particular what changes can reasonably be expected to become
+  part of next releases, and what changes are more likely to be delayed
+  because of lack of resources.
+
+- It helps Coq developers to focus on important axes, and to be more
+  efficient, by making sure that their work will be supported by other
+  developers, including reviewers.
+
+- It helps contributors to know what are the important axes where their
+  contribution is most likely to be welcome and where they can expect
+  the most support from the Coq developers.
+
+- It helps highlighting the axes where more resources are needed, and
+  where the Coq project should try to find more resources.
+
+# Detailed design
+
+This roadmap is a consolidated view created by the Coq developers, based on
+their shared understanding of the priorities of the project and of the
+resources available to work on these priorities. It is completed by a
+discussion with the community, as part of the CEP process.
+
+The fact that an item is part of the roadmap does not mean that its design
+is already fixed, as design can be part of the work to be done. Therefore,
+disagreement can still arise on such items, and may require to be resolved
+through PR reviews, or through the CEP process. However, if an item is
+part of the roadmap, it means that the Coq developers are committed to
+work on it, and that they have resources available to do so, including the
+reviews, so the work should not stall because of lack of resources or lack
+of interest.
+
+The description of each item of the roadmap will include the available
+resources, the expected duration of the work, and the expected outcome,
+including any blocking issues that will need to be resolved to complete
+the work, and any still unresolved questions to be answered.
+
+Rules :
+
+- An item cannot be part of the roadmap if it is not supported by at least
+  two persons, including at least one Coq maintainer to review the changes.
+- No one should be committed to work on more than two items at the same time.
+
+Each Coq Call will start with a roundtable about progress on items of the
+roadmap.
```

Comment:
I suggest having developers break down their multi-month projects into a list of shorter tasks, each of which can ideally be _fully completed_ in a couple of weeks.  Very hard otherwise to tell how much progress is being made.  Items on the list can be checked off in the document as they're completed.

Reply:
Yes. This is a good suggestion. We are slowly changing the culture here, so I'm unwilling to make this a hard requirement (this is already difficult enough to get people to write what they have in mind), but I would definitely want to strive toward that.